adcircpy/mesh/base.py

```diff
@@ -707,3 +689,73 @@ def signed_polygon_area(vertices):
     @lru_cache(maxsize=None)
     def _attributes(self):
         return {}
+
+
+def boundary_indices(elements: [[int]]) -> numpy.array:
+    shape_lengths = {len(element) for element in elements}
+
+    elements = [[element for element in elements
+                 if len(element) == shape_length]
+                for shape_length in shape_lengths]
+
+    boundary_edge_indices = []
+    for shape_elements in elements:
+        shape_elements = numpy.stack(shape_elements, axis=0)
+        shape_length = shape_elements.shape[1]
+        indices = numpy.stack([(shape_elements[:, index],
+                                shape_elements[:, index - shape_length + 1])
+                               for index in range(shape_length)], axis=1).T
+
+        indices = numpy.reshape(
+            indices, (indices.shape[0] * indices.shape[1], indices.shape[2])
+        )
+
+        unique_indices, counts = numpy.unique(indices, axis=0,
+                                              return_counts=True)
+        boundary_edge_indices.extend(unique_indices[counts == 1])
+
+    return boundary_edge_indices
+
+
+def boundary_points(points: numpy.array, elements: [[int]]) -> numpy.array:
+    return numpy.stack(points[boundary_indices(elements)], axis=0)
+
+
+def polygon_from_vertices(vertices: numpy.array) -> MultiPolygon:
+    polygons = polygonize(MultiLineString(vertices.tolist()))
+    polygons = remove_interior_duplicates(polygons)
+    return unary_union(polygons)
+
+
+def remove_interior_duplicates(polygons: [Polygon]) -> [Polygon]:
+    """
+    Given a list of polygons, return a list without polygons whose exteriors represent the interior of another polygon.
+
+    Parameters
+    ----------
+    polygons
+        list of polygons
+
+    Returns
+    -------
+    [Polygon]
+        polygons without duplicates of interiors
+    """
+
+    if type(polygons) is not list:
```

Comment:
This syntax works for some objects, but it will fail in the general case. Use of the built-in `isinstance()` is strongly suggested.